pkg/checkpoint/secret.go

```diff
@@ -19,13 +21,20 @@ func (c *checkpointer) checkpointSecretVolumes(pod *v1.Pod) (*v1.Pod, error) {
 
 	for i := range pod.Spec.Volumes {
 		v := &pod.Spec.Volumes[i]
-		if v.Secret == nil {
-			continue
+
+		if v.Secret != nil {
+			_, err := c.checkpointSecret(pod.Namespace, pod.Name, v.Secret.SecretName, uid, gid)
+			if err != nil {
+				return nil, fmt.Errorf("failed to checkpoint secret for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+			}
 		}
 
-		_, err := c.checkpointSecret(pod.Namespace, pod.Name, v.Secret.SecretName, uid, gid)
-		if err != nil {
-			return nil, fmt.Errorf("failed to checkpoint secret for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		if v.VolumeSource.Projected != nil {
+			p := v.VolumeSource.Projected
+			err := c.checkpointProjectedSecrets(pod.Namespace, pod.Name, v.Name, p.Sources, uid, gid)
```

Comment:
This doesn't checkpoint the configmaps either?

I'd also prefer error'ing if we detect a downward API here. Better to be explicit then create hard to debug scenarios

Reply:
added support for configmaps and error if using downward api